Jenkinsfile

```diff
@@ -1,66 +1,59 @@
-#!groovy
-
-node {
-  checkout scm
-
-  def dockerRepoName = 'zooniverse/panoptes-jenkins'
-  def dockerImageName = "${dockerRepoName}:${BRANCH_NAME}"
-  def newImage = null
-
-  stage('Build Docker image') {
-    newImage = docker.build(dockerImageName)
-    newImage.push()
-  }
-
-  if (BRANCH_NAME == 'master') { // TODO: change this to "master"
-    stage('Update latest tag') {
-      newImage.push('latest')
+pipeline {
+  agent any
+
+  stages {
+    stage('Build Docker image') {
+      steps {
+        script {
+          def dockerRepoName = 'zooniverse/panoptes-jenkins'
+          def dockerImageName = "${dockerRepoName}:${BRANCH_NAME}"
+          def newImage = docker.build(dockerImageName)
+          newImage.push()
+
+          if (BRANCH_NAME == 'master') {
+            stage('Update latest tag') {
+              newImage.push('latest')
+            }
+          }
+        }
+      }
     }
 
     stage('Build staging AMIs') {
-      parallel api: {
-        sh """
-          cd "/var/jenkins_home/jobs/Zooniverse GitHub/jobs/operations/branches/master/workspace" && \
-          ./rebuild.sh panoptes-api-staging
-        """
-      },
-      worker: {
-        sh """
-          cd "/var/jenkins_home/jobs/Zooniverse GitHub/jobs/operations/branches/master/workspace" && \
-          ./rebuild.sh panoptes-dumpworker-staging
-        """
+      when {
+        anyOf { branch 'master'; branch 'fix_jenkinsfile' }
+      }
+      failFast true
+      parallel {
+        stage('Build API') {
+          steps {
+            build job: '/Build Panoptes Staging AMI'
```

Comment:
I want to phase out the old jobs, so I'd rather move their content into this file instead of just triggering them from here. The actual logic of building/deploying should be in this repo (if that means we need an operations Docker image or something, to make things cleaner, then we can do that).

Reply:
understand - we need access to the operations repo so having a docker image would be neat or we can look into another SCM checkout to gain access like the jobs currently do.

Reply:
Yeah, the operations repo has a jenkins file that just does `checkout scm`, so the copy of the repo at `/var/jenkins_home/jobs/Zooniverse GitHub/jobs/operations/branches/master/workspace` is already aways up to date. Docker would obviously be cleaner than having to `cd` to there first.

Reply:
That's weird. I just manually re-ran the operations job and it checked it out to there. Maybe Jenkins cleans up the workspaces for jobs that haven't run in a while? In which case a docker image is definitely the way to go.